utils/mk-consul-env

```diff
@@ -81,8 +81,15 @@ sed -r -e "s/^(NODE=).*/\1$(node-name)/" \
     sudo tee $ENV_FILE >/dev/null
 
 if [[ -n $join_addr ]]; then
-    sudo sed -r -e "s/^(JOIN=).*/\1${join_addr:+-retry-join $join_addr}/" \
-             -i $ENV_FILE
+    # join_addr is not empty and it contains one or more ip addresses separated
+    # by space.
+    # Multiple server addresses must turn into multiple '-retry-join ipN'.
+    # For instance:
+    # '127.0.0.1 192.168.0.1' => '-retry-join 127.0.0.1 -retry-join 192.168.0.1'
+
+    retry_str="$(echo "$join_addr" | tr ' ' '\n' | \
```

Comment:
I think `join_addr` whill not have both the addresses because above we are shifting by 2 thus in case of multiple join ipaddresses but just single `--join` option will end up `join_addr` to have just 1 ipaddress, something like this,
```
[root@ssc-vm-1623:root] hctl bootstrap --mkfs /root/cluster-3-1-ios-ctrl-fail-node-fail.yaml
2021-07-01 16:34:24: Generating cluster configuration... OK
10.230.250.53 10.230.250.48
2021-07-01 16:34:27: Starting Consul server on this node...--join "10.230.250.53 10.230.250.48 "
"10.230.250.53 <<<<<< **join_addr** in mk-consul-env
........ OK
```
I think we need to mention `--join` option multiple times and append ipaddresses to `join_addr`

consider this,

```
[root@ssc-vm-1623:root] diff /home/520478/cortx-hare-fork/cortx-hare-fork-new/cortx-hare-1/utils/hare-bootstrap /opt/seagate/cortx/hare/libexec/hare-bootstrap
284c284,288
< read _ join_peers <<< $(get_server_nodes | grep -vw $(node-name))
---
> # read _ join_peers <<< $(get_server_nodes | grep -vw $(node-name))
> join_peers="$(get_server_nodes | grep -vw "$(node-name)" | awk '{print $2}' | \
>               tr '\n' ' ' || true)"
> # Remove extra trailing spaces
> join_peers=$(echo $join_peers | sed 's/ *$//g')
317c321,322
< join_peers_opt="${join_peers:+--join $join_peers}"
---
> # join_peers_opt="${join_peers:+--join $join_peers}"
> join_peers_opt="${join_peers:+$join_peers}"
319c324,326
< mk-consul-env --mode server --bind $join_ip $join_peers_opt \
---
> join_str="$(echo "$join_peers_opt" | tr ' ' '\n' | \
>                  sed 's/\(^.*$\)/--join \1/' | tr '\n' ' ')"
> mk-consul-env --mode server --bind $join_ip $join_str \
358,364c365,371
<     if (( $count > 5 )); then
<         echo 'Some agent(s) failed to start in due time:' >&2
<         diff <(get_ready_agents | sort) \
<              <(get_all_nodes | awk '{print $1}' | sort) | sed 1d >&2
<         echo 'Check connectivity and firewall (Consul ports must be opened)' >&2
<         exit 1
<     fi
---
> #     if (( $count > 5 )); then
> #         echo 'Some agent(s) failed to start in due time:' >&2
> #         diff <(get_ready_agents | sort) \
> #              <(get_all_nodes | awk '{print $1}' | sort) | sed 1d >&2
> #         echo 'Check connectivity and firewall (Consul ports must be opened)' >&2
> #         exit 1
> #     fi
367c374
<     (( count++ ))
---
> #     (( count++ ))
369a377,378
>
> exit 0
[root@ssc-vm-1623:root] diff /home/520478/cortx-hare-fork/cortx-hare-fork-new/cortx-hare-1/utils/mk-consul-env /opt/seagate/cortx/hare/libexec/mk-consul-env
64c64
<         -j|--join)           join_addr=$2; shift 2 ;;
---
>         -j|--join)           join_addr+=$2; join_addr+=' '; shift 2 ;;
82a83,84
> # Remove extra trailing spaces
> join_addr=$(echo $join_addr | sed 's/ *$//g')
84,85c86,90
<     sudo sed -r -e "s/^(JOIN=).*/\1${join_addr:+-retry-join $join_addr}/" \
<              -i $ENV_FILE
---
>     # sudo sed -r -e "s/^(JOIN=).*/\1${join_addr:+-retry-join $join_addr}/" \
>     #          -i $ENV_FILE
>     retry_str="$(echo "$join_addr" | tr ' ' '\n' | \
>                      sed 's/\(^.*$\)/-retry-join \1/' | tr '\n' ' ')"
>     sudo sed -r -e "s/^(JOIN=).*/\1$retry_str/" -i $ENV_FILE
```

Reply:
Addressed. @mssawant Can you please have a look once again?